### DISCUSSION TITLE
How to add table comments at migrations ?

### DISCUSSION THREAD

**Discussion by hihuangwei** [laravel/framework#32840] [General]
- Laravel Version: 7.11.0
- PHP Version: 7.4.5
- Database Driver & Version: MySQL 8.0.20

### Description:
 How to add table comments at migrations ?

**Comment by thisismhp** [comment]
in migration file add `DB::statement("ALTER TABLE `foo` comment 'comment'")` to `up` method
```
Schema::create("foo", function (Blueprint $table) 
    {
     .
     .
     .
    });


DB::statement("ALTER TABLE `foo` comment 'comment'");

```

**Comment by sergeByishimo** [comment]
Hi @thisismhp 

https://laravel.com/docs/7.x/migrations#column-modifiers

look for comment in the Column Modifiers table.

**Comment by thisismhp** [comment]
Hi @sergeByishimo  
@hihuangwei wants add comment for `tables` not columns

**Comment by sergeByishimo** [comment]
yeah sorry! @thisismhp that answer you provided is right!

i wonder if a much nicer way is PR worthy!!
`$table->comment('example comment')`

**Reply by SamAsEnd** [reply]
> yeah sorry! @thisismhp that answer you provided is right!
> 
> i wonder if a much nicer way is PR worthy!!
> `$table->comment('example comment')`

I prefer `$table->comment = 'example comment'` to make it consistent with other table options like `$table->engine = 'InnoDB'`

**Comment by thisismhp** [comment]
@sergeByishimo :) tnx

**Comment by thisismhp** [comment]
> 
> 
> yeah sorry! @thisismhp that answer you provided is right!
> 
> i wonder if a much nicer way is PR worthy!!
> `$table->comment('example comment')`

this is a feature request.
laravel team must answer it!

**Comment by driesvints** [comment]
Moved to discussions.

**Comment by shanezhiu** [comment]
May `Schema::comment()`  be more  standardized.

**Comment by chuoke** [comment]
Hi, Is this supported?

**Reply by tapacko88** [reply]
No, for now only DB::statement

**Comment by RbusT** [comment]
Table comments are now available from Version 9.14 on. See this [PR](https://github.com/laravel/framework/pull/42401) and the [Blog article](https://laravel-news.com/laravel-9-14-0)
```php
Schema::table('posts', function (Blueprint $table) {
    $table->comment('This is a comment');
});
```

**Reply by dogeow** [reply]
nice

**Comment by TursunboyevJahongir** [comment]
Before Laravel 9, only columns were allowed to comment. Since laravel 9 the ability to comment on the table itself has been added.
If you would like to add a "comment" to a database table, you may invoke the comment method on the table instance. Table comments are currently only supported by **MySQL** and **Postgres**:

```
    Schema::create('calculations', function (Blueprint $table) {
        $table->comment('Business calculations');
     
        // ...
    });
```
